.github/workflows/main.yml

```diff
@@ -28,6 +28,16 @@ jobs:
           cargo fmt --all --manifest-path $manifest --check
         done
 
+  # Ensure there are no clippy warnings
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: rustup update stable && rustup default stable
+      - run: rustup component add clippy
+      # Only check cargo lib for now
+      - run: cargo clippy -p cargo --lib -- -D warnings
```

Comment:
Should we instead check everything and stop blocking tests on there being no warnings?

Reply:
I am not sure now is the right time to run clippy against all crates. Running `clippy` for everything requires `#![allow(clippy::all)]` to be added to everything in `crates/*`. It would also mean any lints that are set to `warn` would need to be copied around as well. It might be better to wait until [the proposal for `[lints]`](https://github.com/rust-lang/rfcs/pull/3389) is accepted and implemented, and cargo has a workspace of its own.

Reply:
Just now thought of the workaround of running the `check` command instead.  Overall though, this was a suggestion and we can do this later.